packages/node_modules/@ciscospark/internal-plugin-user/src/user.js

```diff
@@ -49,27 +50,43 @@ const User = SparkPlugin.extend({
    * @param {Object} options.confirmationCode (required -- optional if verification token is provided)
    * @param {Object} options.id (required -- optional if verification token is provided)
    * @param {Object} options.verificationToken (required -- optional if uuid and verification token provided)
+   * @param {Object} options.email (required with verificationToken for Federation/global user)
    * @returns {Promise} Resolves with a userSession
    */
   activate(options = {}) {
     if (!(options.verificationToken || (options.confirmationCode && options.id))) {
       return Promise.reject(new Error('either options.verificationToken is required or both options.confirmationCode and options.id are required'));
     }
+
+    // Email is required for federation
+    if (this.spark.config.credentials.federation && !options.email) {
+      return Promise.reject(new Error('email is required with federation option'));
+    }
+
     options.scope = this.spark.config.credentials.scope;
-    return this.request({
-      uri: this.config.activationUrl,
-      method: 'POST',
-      body: options,
-      auth: {
-        user: this.spark.config.credentials.client_id,
-        pass: this.spark.config.credentials.client_secret,
-        sendImmediately: true
-      }
-    })
-      .then((res) => {
-        this.spark.credentials.set({supertoken: res.body.tokenData});
-        return res.body;
-      });
+
+    // if we have options.email and options.verificationToken
+    // and Federation flag is enabled, flag that we need to
+    // lookup user's CI.
+    const needsCILookup = this.spark.config.credentials.federation && options.email && options.verificationToken;
+    return this.getUserCI(options.email, needsCILookup).then(() => {
+      const activateOptions = Object.assign({}, options);
+      delete activateOptions.email;
```

Comment:
Why delete `email` here?

Reply:
The API call takes all the same passed-in args but it chokes on having the email included.

Reply:
👌